specific/cmo_hr/models/hr.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+import datetime
+
+from openerp import fields, models, api, _
+from openerp.exceptions import ValidationError
+
+
+class HrExpenseExpense(models.Model):
+    _inherit = 'hr.expense.expense'
+
+    employee_request_id = fields.Many2one(
+        'hr.employee',
+        string='Employee Request',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+        default=lambda self: self._default_employee_request_id(),
+    )
+    request_date = fields.Date(
+        string='Request Date',
+        default=lambda self: fields.Date.context_today(self),
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    due_date = fields.Date(
+        string='Due Date',
+        default=lambda self: fields.Date.context_today(self),
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    payment_by = fields.Selection(
+        [('cash', 'Cash'),
+         ('cashier_cheque', 'Cashier Cheque'),
+         ('bank_transfer', 'Bank Transfer'),
+         ('ac_payee', 'A/C Payee'),
+        ],
+        string='Payment By',
+    )
+    bank_transfer_ref = fields.Text(
+        string='Bank Transfer Ref.',
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    ac_payee_ref = fields.Text(
+        string='A/C Payee Ref.',
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    state = fields.Selection(
+        selection=[
+            ('draft', 'New'),
+            ('validate', 'Waiting Validate'),
+            ('cancelled', 'Refused'),
+            ('confirm', 'Waiting Approval'),
+            ('accepted', 'Approved'),
+            ('done', 'Waiting Payment'),
+            ('paid', 'Paid'),
+        ],
+    )
+    employee_id = fields.Many2one(
+        states={
+            'draft': [('readonly', True)],
+            'confirm': [('readonly', True)],
+        },
+    )
+    department_id = fields.Many2one(
+        states={
+            'draft': [('readonly', True)],
+            'confirm': [('readonly', True)],
+        },
+    )
+
+    @api.model
+    def create(self, vals):
+        ctx = self._context.copy()
+        current_date = datetime.date.today()
+        fiscalyear_id = self.env['account.fiscalyear'].find(dt=current_date)
+        ctx["fiscalyear_id"] = fiscalyear_id
+        if (not ctx.get('is_employee_advance', False)) and \
+                ctx.get('number', '/') == '/':
+            vals['number'] = self.env['ir.sequence']\
+                .with_context(ctx).get('cmo.expense')
+        elif ctx.get('is_employee_advance', False) and \
```

Comment:
Better to use module Doctype for HR Expense / Advance
Please come to me to talk about it.